backend/planty/dashboard/views.py

```diff
@@ -290,3 +295,226 @@ def post(self, request: Request):
         plant.save()
 
         return Response(status=status.HTTP_200_OK)
+
+
+class InstructionsView(APIView):
+    def get(self, request: Request):
+        user: User = request.user
+        my_instructions = Instruction.objects.filter(user=user, public=False)
+
+        my_instructions_json = []
+        for instruction in my_instructions:
+            instruction_json = {
+                'id': str(instruction.id),
+                'name' : instruction.name,
+                'species': instruction.species,
+
+                'watering': instruction.watering,
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+            }
+            my_instructions_json.append(instruction_json)
+
+        return Response(my_instructions_json, status=status.HTTP_200_OK)
+
+    def post(self, request: Request):
+        user: User = request.user
+        serializer = InstructionCreateSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        instruction: Instruction = Instruction.objects.create(
+            id=uuid4(),
+            name=name,
+
+            user=user,
+            species=data['species'],
+
+            watering=data['watering'],
+            insolation=data['insolation'],
+            fertilizing=data['fertilizing']
+        )
+        instruction.save()
+
+        return Response(status=status.HTTP_201_CREATED)
+
+    def put(self, request: Request):
+        user: User = request.user
+        serializer = InstructionSelectSerializer(data=self.kwargs['uuid'])
```

Comment:
Po pierwsze, mi się wydaje, że argument jest w `self.kwargs['id']`, `uuid` to raczej wskazanie typu. Dlatego też raczej nie jest potrzebne przepuszczenie tego przez serializer'a, bo jeśli nie będzie to poprawne UUID, to po prostu nie zmatch'uje tego url'a. Natomiast trzeba sprawdzić czy istnieje obiekt pod tym UUID tak jak robisz niżej. Tego powinno wystarczyć

Reply:
https://docs.djangoproject.com/en/4.0/topics/http/urls/

Reply:
okej, teraz zrozumiałem czym jest 'uuid'
zmieniłem na 'self.kwargs['id']'
usunąłem tego serializer'a